parity/modules.rs

```diff
@@ -112,6 +113,37 @@ fn sync_arguments(io_path: &str, sync_cfg: SyncConfig, net_cfg: NetworkConfigura
 	BootArgs::new().stdin(service_payload).cli(cli_args)
 }
 
+#[cfg(feature="stratum")]
+pub fn stratum (hypervisor_ref: &mut Option<Hypervisor>, maybe_config: &Option<::ethcore::miner::StratumOptions>)
+{
+	use ethcore_stratum;
+
+	if let &Some(ref config) = maybe_config {
+		let mut hypervisor = hypervisor_ref.take().expect("There should be hypervisor for ipc configuration");
```

Comment:
requiring that the function take `&mut Hypervisor` and `&StratumOptions` instead of `Option`s will avoid these checks.